modules/gateway/peers.go

```diff
@@ -175,6 +175,9 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 	if addr == g.Address() {
 		return errors.New("can't connect to our own address")
 	}
+	if build.Release != "testing" && addr.IsLoopback() {
```

Comment:
not really what I meant, but that's an interesting idea. I was thinking more along the lines of "if you mock an object during testing, you can implement the testing behavior inside the object instead of checking for `"testing"` inside the real functions."